src/installJava.ts

```diff
@@ -23,21 +23,23 @@ export function installJava({
   javaVersion,
   jabbaVersion = defaultJabbaVersion,
   outputChannel,
-}: InstallJavaOptions): Promise<string> {
+}: InstallJavaOptions): Promise<string | undefined> {
```

Comment:
See my suggestion below, I think this can stay as it was if we rethrow the error and make the Promise fail

```suggestion
}: InstallJavaOptions): Promise<string> {
```